oss-internship-2020/pffft/main_pffft_sandboxed.cc

```diff
@@ -0,0 +1,203 @@
+        .AllowOpen()
+        .AllowRead()
+        .AllowWrite()
+        .AllowSystemMalloc()
+        .AllowExit()
+        .AllowSyscalls({
+            __NR_futex,
+            __NR_close,
+            __NR_getrusage,
+        })
+        .BuildOrDie();
+  }
+};
+
+double UclockSec() { return static_cast<double>(clock()) / CLOCKS_PER_SEC; }
+
+int array_output_format = 0;
+
+void ShowOutput(const char* name, int n, int cplx, float flops, float t0,
+                float t1, int max_iter) {
+  float mflops = flops / 1e6 / (t1 - t0 + 1e-16);
+  if (array_output_format) {
+    if (flops != -1) {
+      printf("|%9.0f   ", mflops);
+    } else
+      printf("|      n/a   ");
+  } else {
+    if (flops != -1) {
+      printf("n=%5d, %s %16s : %6.0f MFlops [t=%6.0f ns, %d runs]\n", n,
+             (cplx ? "CPLX" : "REAL"), name, mflops,
+             (t1 - t0) / 2 / max_iter * 1e9, max_iter);
+    }
+  }
+  fflush(stdout);
+}
+
+absl::Status PffftMain() {
+  PffftSapiSandbox sandbox;
+  SAPI_RETURN_IF_ERROR(sandbox.Init());
+
+  return absl::OkStatus();
+}
+
+int main(int argc, char* argv[]) {
+  // Initialize Google's logging library.
+  google::InitGoogleLogging(argv[0]);
+
+  gflags::ParseCommandLineFlags(&argc, &argv, true);
+
+  // kTransformSizes is a vector keeping the values by which iterates n, its value
+  // representing the input length. More concrete, n is the number of
+  // data points the caclulus is up to (determinating its accuracy).
+  // To show the performance of Fast-Fourier Transformations the program is
+  // testing for various values of n.
+  constexpr int kTransformSizes[] = {64,    96,     128,        160,         192,     256,
+                   384,   5 * 96, 512,        5 * 128,     3 * 256, 800,
+                   1024,  2048,   2400,       4096,        8192,    9 * 1024,
+                   16384, 32768};
+
+  LOG(INFO) << "Initializing sandbox...\n";
+
+  PffftSapiSandbox sandbox;
+  absl::Status init_status = sandbox.Init();
+
+  if (absl::Status status = PffftMain(); !status.ok()) {
+    LOG(ERROR) << "Initialization failed: " << status.ToString();
+    return EXIT_FAILURE;
+  }
+
+  LOG(INFO) << "Initialization: " << init_status.ToString();
+
+  PffftApi api(&sandbox);
+  int cplx = 0;
+
+  do {
+    for (int n : kTransformSizes) {
+      const int n_float = n * (cplx ? 2 : 1);
+      int n_bytes = n_float * sizeof(float);
+
+      std::vector<float> work(2 * n_float + 15, 0.0);
+      sapi::v::Array<float> work_array(&work[0], work.size());
+
+      float x[n_bytes], y[n_bytes], z[n_bytes];
+      sapi::v::Array<float> x_array(x, n_bytes), y_array(y, n_bytes), z_array(z, n_bytes);
+
+      double t0;
+      double t1;
+      double flops;
+
+      int k;
+      int max_iter = 5120000 / n * 4;
+
+      for (k = 0; k < n_float; ++k) {
+        x[k] = 0;
+      }
+
+      // FFTPack benchmark
+      {
+        // SIMD_SZ == 4 (returning value of pffft_simd_size())
+        int max_iter_ = max_iter / 4;
```

Comment:
`max_iter_` would be a name for a class member. It's also a poor choice as it's almost the same as `max_iter` and there is no indication in the name why they should be different.
How about `simd_size_iter`?